src/js/server/inline.js

```diff
@@ -52,20 +52,19 @@ fluid.defaults("gpii.handlebars.inlineTemplateBundlingMiddleware.request", {
 });
 
 gpii.handlebars.inlineTemplateBundlingMiddleware.loadTemplates =  function (that) {
-    var resolvedTemplateDirs = gpii.express.hb.resolveAllPaths(that.options.templateDirs);
-
     // Clear out the existing template content, as we might also be called during a "reload".
     fluid.each(["layouts", "pages", "partials"], function (key) {
         that.templates[key] = {};
     });
 
+    var resolvedTemplateDirs = gpii.express.hb.resolveAllPaths(that.options.templateDirs);
     fluid.each(resolvedTemplateDirs, function (templateDir) {
         // Start with each "views" directory and work our way down
         var dirContents = fs.readdirSync(templateDir);
         dirContents.forEach(function (entry) {
             var subDirPath = path.resolve(templateDir, entry);
             var stats = fs.statSync(subDirPath);
-            if (stats.isDirectory() && that.options.allowedTemplateDirs.indexOf[entry] !== -1) {
```

Comment:
ouch!

Reply:
Yes, I was pretty stunned to find that one, which has apparently been there for ages.  For whatever reason, using square brackets on any function returns `undefined` rather than an error (try `"this".indexOf[2]` in a browser console).  The net result of this goof was that any subdirectory would have been scanned, but we have managed to avoid picking up wild cruft so far, likely because of the additional check for the file's extension.

Although there could be many variations on this lurking around, I did confirm that of the repos I have checked out, the specific `indexOf[]` pattern is only found in this file.